### PR TITLE
feat: integrate usage collection

### PR DESCRIPTION
**Pull request by akaJSaunders**
### JIRA

https://companieshouse.atlassian.net/browse/SR-79

### Change Description

Config changes: https://github.com/companieshouse/chs-configs/pull/1275

### Instructions
1. add the following environment variables to test it locally:

```
PIWIK_SITE_ID=13
PIWIK_URL=https://matomo.platform.aws.chdev.org
PIWIK_LANDING_PAGE_START_GOAL_ID=10
```

2. go to https://matomo.platform.aws.chdev.org/index.php?module=CoreHome&action=index&date=yesterday&period=day&idSite=13#?idSite=13&period=day&date=yesterday&segment=&category=Dashboard_Dashboard&subcategory=1

### Work Checklist

- [ ] Tests added where applicable
- [ ] Test coverage of new code meets target of 80%
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

---

### Merge Instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/modules/config-handler/ConfigValidation.schema.ts [hunk 1]
```diff
@@ -12,7 +12,9 @@ export const configValidationSchema = Joi.object({
   CACHE_SERVER: Joi.string().required(),
   COOKIE_EXPIRATION_IN_SECONDS: Joi.string().required(),
   DOCUMENT_AMENDMENT_FEE: Joi.number().required(),
-  PROCESSING_DELAY_EVENT: Joi.string()
+  PROCESSING_DELAY_EVENT: Joi.string(),
+  PIWIK_URL: Joi.string(),
```
Comment (rhysbarrett) [inline src/modules/config-handler/ConfigValidation.schema.ts:16]: Should these new config vars be `required()`?

Comment (rhysbarrett) [reply src/modules/config-handler/ConfigValidation.schema.ts:16]: Never mind, I can see your `if piwik` in the Nunjucks template. I'm curious though about which environments piwik won't be configured in?

File: src/views/layout.njk [hunk 2]
```diff
@@ -42,4 +42,17 @@
...
+  {% if piwik %}
+    <script src="//{{ cdn.host }}/javascripts/app/piwik-enable.js"></script>
+    <script type="application/javascript">
+      (function() {
+        bindPiwikListener('suppressions', '{{ piwik.url }}', {{ piwik.siteId }});
+      }());
+    </script>
+    <noscript>
+      <p>
+        <img src="{{ piwik.url }}/piwik.php?idsite={{ piwik.siteId }}" style="border:0;" alt="" />
```
Comment (rhysbarrett) [inline src/views/layout.njk:54]: Should we populate the `alt` attribute, for accessibility purposes?

**Review by rhysbarrett — COMMENTED** [review]
Could you populate the "Change Description" section on the PR and maybe detail how you tested this so others can test as well?

File: src/views/start.njk [hunk 3]
```diff
@@ -50,7 +50,12 @@
         govukButton({
           text: 'Start now',
           classes: 'govuk-!-margin-top-2 govuk-!-margin-bottom-8',
-          isStartButton: true
+          isStartButton: true,
```
Comment (rhysbarrett) [inline src/views/start.njk:53]: Did you get an answer to your question about whether we need to apply the tracking everywhere or just the Start Now button?

File: src/app.ts [hunk 4]
```diff
@@ -40,6 +40,13 @@ app.set('port', getConfigValue('PORT'));
 // where nunjucks templates should resolve to
 const viewPath = path.join(__dirname, 'views');
 
+// set up usage statistics
+
```
Comment (n-r-b) [inline src/app.ts:44]: please remove this space

File: src/app.ts [hunk 5]
```diff
@@ -40,6 +40,13 @@ app.set('port', getConfigValue('PORT'));
...
+app.locals.piwik = {
```
Comment (n-r-b) [inline src/app.ts:45]: if the `url` and `siteId` are not configured (undefined), would we still want to add `piwik` to `app.locals`?

File: src/modules/config-handler/ConfigValidation.schema.ts (hunk 1 above)
Comment (akaJSaunders) [reply src/modules/config-handler/ConfigValidation.schema.ts:16]: I believe docker

File: src/views/layout.njk (hunk 2 above)
Comment (akaJSaunders) [reply src/views/layout.njk:54]: this doesnt change the content of the pages I believe

File: src/views/start.njk (hunk 3 above)
Comment (akaJSaunders) [reply src/views/start.njk:53]: no

File: src/app.ts (hunk 5 above)
Comment (akaJSaunders) [reply src/app.ts:45]: we have a check in the templates for when piwik is not set

File: src/views/start.njk (hunk 3 above)
Comment (akaJSaunders) [reply src/views/start.njk:53]: seems to track throughout all pages

File: src/modules/config-handler/ConfigValidation.schema.ts [hunk 6]
```diff
@@ -12,7 +12,10 @@ export const configValidationSchema = Joi.object({
...
+  PIWIK_SITE_ID: Joi.string(),
+  PIWIK_LANDING_PAGE_START_GOAL_ID: Joi.string()
```
Comment (n-r-b) [inline src/modules/config-handler/ConfigValidation.schema.ts]: is this always a number?

File: src/modules/config-handler/ConfigValidation.schema.ts [hunk 7]
```diff
@@ -12,7 +12,10 @@ export const configValidationSchema = Joi.object({
   CACHE_SERVER: Joi.string().required(),
   COOKIE_EXPIRATION_IN_SECONDS: Joi.string().required(),
   DOCUMENT_AMENDMENT_FEE: Joi.number().required(),
-  PROCESSING_DELAY_EVENT: Joi.string()
+  PROCESSING_DELAY_EVENT: Joi.string(),
+  PIWIK_URL: Joi.string().required(),
+  PIWIK_SITE_ID: Joi.string().required(),
+  PIWIK_LANDING_PAGE_START_GOAL_ID: Joi.string().required()
```
Comment (n-r-b) [inline src/modules/config-handler/ConfigValidation.schema.ts:18]: is this always a number?

File: src/views/start.njk [hunk 8]
```diff
@@ -50,7 +50,12 @@
...
+          attributes: {
+            id: 'start-now',
+            'data-event-id': 'start-now-button' if piwik,
+            onclick: '_paq.push(["trackGoal", ' + piwik.startGoalID + ']);'
```
Comment (n-r-b) [inline src/views/start.njk:57]: should we check `if piwik` exists first before getting the goal id?

File: src/modules/config-handler/ConfigValidation.schema.ts (hunk 7 above)
Comment (akaJSaunders) [reply src/modules/config-handler/ConfigValidation.schema.ts:18]: yes, but its being added to a string: `onclick: '_paq.push(["trackGoal", ' + piwik.startGoalID + ']);'`

**Review by n-r-b — APPROVED** [review]

**Pull request merged**